.github/actions/rust/action.yml

```diff
@@ -20,11 +20,21 @@ runs:
         toolchain: ${{ inputs.version }}
         components: ${{ inputs.components }}
 
-    - name: Use sccache
-      uses: mozilla-actions/sccache-action@v0.0.4
+    - name: Install cargo-binstall (Windows)
+      if: runner.os == 'Windows'
+      shell: pwsh
+      run: Set-ExecutionPolicy Unrestricted -Scope Process; iex (iwr "https://raw.githubusercontent.com/cargo-bins/cargo-binstall/main/install-from-binstall-release.ps1").Content
 
-    - name: Enable sscache
+    - name: Install cargo-binstall (Windows)
+      if: runner.os != 'Windows'
       shell: bash
-      run: |
-        echo "SCCACHE_GHA_ENABLED=true" >> "$GITHUB_ENV"
-        echo "RUSTC_WRAPPER=sccache" >> "$GITHUB_ENV"
+      run: curl -L --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/cargo-bins/cargo-binstall/main/install-from-binstall-release.sh | bash
+
+    - name: Install Rust tools
+      shell: bash
+      run: cargo +${{ inputs.version }} binstall --no-confirm cargo-llvm-cov cargo-nextest flamegraph
+
+    - name: Use Rust cache
+      uses: Swatinem/rust-cache@v2
+      with:
+        cache-all-crates: "true"
```

Comment:
Looking at the docs, `cache-targets` seems like it is where the performance wins come from, but I'm concerned that we're signing on for very large caches.